src/components/MainPage.js

```diff
@@ -59,48 +56,70 @@ type Props = {
 
 type State = {
   inputValue: number | string,
-  favorites: Array<number>,
+  activePage: number,
+  favorites: string,
 };
 
 class MainPage extends Component<Props, State> {
   componentDidMount() {
-    this.props.getMovies.movies();
+    this.props.getMovies.movies(null, 1);
+    this.setState({ activePage: 1 });
+    localStorage.setItem('favorites', 'init');
   }
 
   handleSearchChange = (event: SyntheticInputEvent<HTMLInputElement>) => {
     this.setState({ inputValue: event.target.value });
+    this.setState({ activePage: 1 });
   };
 
   handleSubmit = (event: SyntheticInputEvent<HTMLInputElement>) => {
     event.preventDefault();
-    this.props.getMovies.movies(this.state.inputValue);
+    this.props.getMovies.movies(this.state.inputValue, this.state.activePage);
   };
 
-  //TODO: add functionality to handle list of favorites movies and bottom navigation
-  handleBottomNavigationChange = (
-    event: SyntheticInputEvent<HTMLInputElement>
-  ) => {
-    event.preventDefault();
-    console.log('handleBottomNavigationChange');
-  };
-
-  openListOfFavorites = (event: SyntheticInputEvent<HTMLInputElement>) => {
-    event.preventDefault();
-    console.log('openListOfFavorites');
+  handlePageChange = pageNumber => {
+    const value = this.state.inputValue;
+    if (value) {
+      this.setState({ activePage: pageNumber });
+      this.props.getMovies.movies(value, pageNumber);
+    } else {
+      this.setState({ activePage: pageNumber });
+      this.props.getMovies.movies(null, pageNumber);
+    }
   };
 
-  addMovieToListOfFavorites = () => {
-    console.log('addMovieToListOfFavorites');
+  addMovieToListOfFavorites = id => {
+    let favorites = localStorage.getItem('favorites');
+    if (favorites) {
+      let res = favorites.split(',');
+      let duplicate = res.find(function(element) {
+        return element === id.toString();
+      });
+      if (!duplicate) {
+        res.push(id.toString());
+        let favToString = res.join(',');
+        localStorage.setItem('favorites', favToString);
+      }
+    }

```

Comment:
Ok, let's try functional composition here.

Pls, replace this part with helper with `JSON.parse`, `flow`, `find` from lodash/fp.

Reply:
Done